orttraining/orttraining/include/runner/orttraining_c_api.h

```diff
@@ -0,0 +1,308 @@
+  ORT_TRAINING_USE_GIST = 0,
+  ORT_TRAINING_USE_CUDA = 1,
+  ORT_TRAINING_USE_PROFILER = 2,
+  ORT_TRAINING_USE_TENSORBOARD = 3,
+  ORT_TRAINING_IS_PERFTEST = 4,
+  ORT_TRAINING_SHUFFLE_DATA = 5
+};
+
+typedef enum OrtDataUse {
+  ORT_DATAUSE_TRAINING = 0,
+  ORT_DATAUSE_TESTING = 1
+};
+
+struct OrtValueCollection;
+
+typedef void(__stdcall* OrtErrorFunctionCallback)(_In_ const size_t ulCount, _In_ OrtValueCollection* output);  // Array of outputs (e.g. label, predictions, loss).
+typedef void(__stdcall* OrtEvaluationFunctionCallback)(size_t num_samples, size_t step);
+typedef void(__stdcall* OrtDataGetBatchCallback)(_In_ const size_t ulBatchSize, _In_ const size_t ulCount, _In_ OrtValueCollection* data); // Array of input locations, then set by call-back implemenation by calling SetAt.
+
+
+// The actual types defined have an Ort prefix
+//ORT_RUNTIME_CLASS(OrtValueCollection);
+struct OrtValueCollection {
+  std::vector<OrtValue*> m_rgValues;
+  std::vector<std::string> m_rgNames;
+
+  OrtValueCollection(int nCapacity) : m_rgValues(nCapacity), m_rgNames(nCapacity) {}
+  ~OrtValueCollection() {}
+
+  size_t Capacity() {
+    return m_rgValues.capacity();
+  }
+
+  size_t Count() {
+    return m_rgValues.size();
+  }
+
+  std::vector<OrtValue*> ValuePtrs() {
+    return m_rgValues;
+  }
+
+  std::vector<std::string> Names() {
+    return m_rgNames;
+  }
+
+  void BeforeUsingAsInput(int queue_id) {
+    for (size_t i = 0; i < m_rgValues.size(); i++) {
+      if (m_rgValues[i]->Fence())
+        m_rgValues[i]->Fence()->BeforeUsingAsInput(onnxruntime::kCpuExecutionProvider, queue_id);
+    }
+  }
+};
+
+struct OrtTrainingApi;
+typedef struct OrtTrainingApi OrtTrainingApi;
+
+
+// The actual types defined have an Ort prefix
+//ORT_RUNTIME_CLASS(TrainingParameters);
+struct OrtTrainingParameters {
+  struct onnxruntime::training::TrainingRunner::Parameters m_param;
+  std::string m_strInputLabels;
+  std::string m_strOutputPredictions;
+  std::string m_strOutputLoss;
+  std::string m_strLossFunction;
+  char* m_pszInitFeedNames;
+  bool m_bUseCuda;
+  bool m_bUseTensorboard;
+  OrtDataGetBatchCallback m_fnTrainingDataGetBatch;
+  OrtDataGetBatchCallback m_fnTestingDataGetBatch;
+  std::vector<std::string> m_rgstrDataFeedNames;
+  OrtValueCollection* m_pTrainingData;
+  OrtValueCollection* m_pTestingData;
+  onnxruntime::training::TrainingRunner* m_pTrainingRunner;
+  onnxruntime::training::IDataLoader* m_pTrainingDataLoader;
+  onnxruntime::training::IDataLoader* m_pTestingDataLoader;
+
+  OrtTrainingParameters() {
+    m_bUseCuda = false;
+    m_bUseTensorboard = true;
+    m_pTrainingData = nullptr;
+    m_pTestingData = nullptr;
+    m_fnTrainingDataGetBatch = nullptr;
+    m_fnTestingDataGetBatch = nullptr;
+    m_pTrainingRunner = nullptr;
+    m_pTrainingDataLoader = nullptr;
+    m_pTestingDataLoader = nullptr;
+    m_pszInitFeedNames = nullptr;
+  }
+
+  ~OrtTrainingParameters() {
+    if (m_pTrainingDataLoader != nullptr)
```

Comment:
>m_pTrainingDataLoader [](start = 8, length = 21)

The class has dynamically allocated memory but does not provide or deletes copy ctor or assignment operator